tests/end_to_end/end_to_end.cpp

```diff
@@ -1956,15 +1956,19 @@ _wrong_map_reuse_test(ebpf_execution_type_t execution_type)
     REQUIRE(bpf_obj_pin(port_map_fd, "/ebpf/global/outer_map") == 0);
 
     // Open eBPF program file.
-    bpf_object* object = bpf_object__open(file_name);
-    REQUIRE(object != nullptr);
-    bpf_program* program = bpf_object__next_program(object, nullptr);
+    bpf_object_ptr object;
+    {
+        bpf_object* local_object = bpf_object__open(file_name);
+        REQUIRE(local_object != nullptr);
+        object.reset(local_object);
+    }
+    bpf_program* program = bpf_object__next_program(object.get(), nullptr);
     REQUIRE(program != nullptr);
 
     // Try to load the program.  This should fail because the maps can't be reused.
-    REQUIRE(bpf_object__load(object) == -EINVAL);
+    REQUIRE(bpf_object__load(object.get()) == -EINVAL);
 
-    bpf_object__close(object);
+    bpf_object__close(object.release());
```

Comment:
Won't this already happen when `object` goes out of scope?
If you keep this line will it get double-decremented when it goes out of scope?

Reply:
Yes it will happen when `object` goes out of scope. Since this `bpf_object__close` call was in the middle of the test case, I jsut assumed that was a requirement of the test to close the object here itself (and not at the end of the test).

Since we called `object.release()`, it will not be freed again.

Reply:
Can you delete this line and see if the tests still pass?  I suspect this line isn't needed.

Reply:
Yes I just now tested that and it works. I am pushing the change.